### PR TITLE
!per #17799 Remove support for non-permanent deletes

### PR DESCRIPTION
**Pull request by patriknw**
**Skip the first commit, it is from PR #17818, which will be merged first.**
- Remove the permanent flag in deleteMessages
- old records stored with deletion flag are still not replayed

### CODE REVIEW

**Comment by patriknw** [comment]
@ktoso I had a local failure in SnapshotStoreSpec. Timing issue (I don't think it was a new problem). See fix in a6fe665

**Comment by patriknw** [comment]
@ktoso ok to merge?

File: akka-persistence/src/main/scala/akka/persistence/snapshot/SnapshotStore.scala [hunk 1]
```diff
@@ -61,12 +62,13 @@ trait SnapshotStore extends Actor with ActorLogging {
...
+      deleteAsync(persistenceId, criteria).map {
         case _ ⇒ DeleteSnapshotsSuccess(criteria)
-      } recover {
+      }.recover {
         case e ⇒ DeleteSnapshotsFailure(criteria, e)
-      } to (self, senderPersistentActor())
-      if (publish) context.system.eventStream.publish(d)
+      }.pipeTo(self)(senderPersistentActor()).onComplete {
+        case _ ⇒ if (publish) context.system.eventStream.publish(d)
+      }
```
Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/snapshot/SnapshotStore.scala]: :)

**Comment by ktoso** [comment]
LGTM

**Comment by patriknw** [comment]
Refs #17799

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala [hunk 2]
```diff
@@ -378,21 +378,8 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
...
-   * they permanently deleted from the journal.
-   *
-   * @param toSequenceNr upper sequence number bound of persistent messages to be deleted.
-   * @param permanent if `false`, the message is marked as deleted, otherwise it is permanently deleted.
-   */
-  def deleteMessages(toSequenceNr: Long, permanent: Boolean): Unit = {
-    journal ! DeleteMessagesTo(persistenceId, toSequenceNr, permanent)
-  }
+  def deleteMessages(toSequenceNr: Long): Unit =
+    deleteMessages(toSequenceNr)
```
Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: oops, refactoring bug
I'll fix
no tests?

**Pull request merged**